backend/corpora/lambdas/api/v1/curation/collections/collection_id/actions.py

```diff
@@ -35,5 +50,57 @@ def get(collection_id: str, token_info: dict):
     return jsonify(collection_response)
 
 
-def patch(collection_id: str, body: dict, token_info: dict):
-    return update_collection(collection_id, body, token_info)
+@dbconnect
+def patch(collection_id: str, body: dict, token_info: dict) -> Response:
+    """
+    Curation API: update the metadata for a Collection. Only included field will be altered. If links are provided, all
+    existing links will be replaced EXCEPT for the doi, which will only be replaced if a new doi is provided. If a bad
+    doi is provided, none of the provided fields are updated, and the original doi (and published metadata) remain.
+    @param collection_id: the id of the Collection
+    @param body: dict with fields to update
+    @param token_info: the user's access token info
+    @return: an http Response
+    """
+    db_session = g.db_session
+
+    new_doi_provided = False
+    if "links" in body:
+        if not body["links"]:
+            raise InvalidParametersHTTPException(detail="If provided, the 'links' array may not be empty")
+        keep_links = False  # links have been provided; replace all old links
+        for link in body["links"]:
+            if link["link_type"] == ProjectLinkType.DOI.name:
+                new_doi_provided = True
+    else:
+        keep_links = True  # links have NOT been provided; keep existing links
+    collection, errors = get_collection_and_verify_body(db_session, collection_id, body, token_info)
+
+    if new_doi_provided:
+        # Compute the diff between old and new DOI
+        old_doi = collection.get_doi()
+        new_doi = normalize_and_get_doi(body, errors)
+        if new_doi and new_doi != old_doi:
+            # If the DOI has changed, fetch and update the metadata
+            publisher_metadata = get_publisher_metadata(new_doi, errors)
+            body["publisher_metadata"] = publisher_metadata
+    else:
+        # re-add original doi link
+        if new_links := body.get("links"):
+            new_links.extend(
+                [
+                    link.to_dict_keep({DbCollectionLink: EntityColumns.link_cols})
+                    for link in collection.links
+                    if link.link_type == CollectionLinkType.DOI
+                ]
+            )
+    if errors:
+        raise InvalidParametersHTTPException(detail=errors)
+
+    collection.update(**body, keep_links=keep_links)
+    collection_dict = collection.to_dict()
```

Comment:
if you use to_dict_keep you wont need to do as much data shuffling below. You'll also avoid converting the whole collection to dict.